src/pages/MissingPersonalDetails/MissingPersonalDetailsMagicCodeModal.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useEffect} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import ValidateCodeActionModal from '@components/ValidateCodeActionModal';
+import useLocalize from '@hooks/useLocalize';
+import {clearDraftValues} from '@libs/actions/FormActions';
+import {clearPersonalDetailsErrors} from '@libs/actions/PersonalDetails';
+import {requestValidateCodeAction} from '@libs/actions/User';
+import {getLatestError} from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type MissingPersonalDetailsMagicCodeModalProps = {
+    handleSubmitForm: (validateCode: string) => void;
+    isValidateCodeActionModalVisible: boolean;
+    onClose?: () => void;
+};
+
+function MissingPersonalDetailsMagicCodeModal({onClose, isValidateCodeActionModalVisible, handleSubmitForm}: MissingPersonalDetailsMagicCodeModalProps) {
+    const {translate} = useLocalize();
+    const [privatePersonalDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const [validateCodeAction] = useOnyx(ONYXKEYS.VALIDATE_ACTION_CODE);
+    const privateDetailsErrors = privatePersonalDetails?.errors ?? undefined;
+    const validateLoginError = getLatestError(privateDetailsErrors);
+    const primaryLogin = account?.primaryLogin ?? '';
+
+    const missingDetails =
+        !privatePersonalDetails?.legalFirstName ||
+        !privatePersonalDetails?.legalLastName ||
+        !privatePersonalDetails?.dob ||
+        !privatePersonalDetails?.phoneNumber ||
+        isEmptyObject(privatePersonalDetails?.addresses) ||
+        privatePersonalDetails.addresses.length === 0;
```

Comment:
Isnt there a helper method for this? i think we should abstract it to one util methodas its reused